### ISSUE TITLE
Feedback on local setup and getting started

### ISSUE THREAD

**Issue by TravisCarden** [phenaproxima/starshot-prototype#126]
It's time to give some dedicated attention to getting set up locally for evaluation and contribution. Only incidental effort has been spent on associated automation and documentation. Please provide your feedback, including what has worked for you and what hasn't, what blockers you encountered, what functionality or documentation you needed but couldn't find, etc. We'll create spinoff issues for discussion and implementation.

**Comment by alexb7217** [comment]
Okay, giving local setup a... "shot".

**Comment by isalmanhaider** [comment]
Why not consider switching to Fish shell? It enhances the command-line experience by providing autocompletion for your most-used and recent commands. Additionally, Fish offers real-time feedback by highlighting command syntax to indicate whether commands are correct or recognized. This can significantly streamline workflows - while you can further amplify the experience with short aliases on top of Fish.

Fish shell is also particularly effective for Git operations. It can automatically suggest Git branches and commands as you type, which simplifies navigating through your repositories and enhances your version control workflow.

While ddev ssh is efficient, it typically requires typing out full commands repeatedly or cycling through command history with the arrow keys. By adopting Fish shell, you can reduce repetitive typing and increase productivity, making it a superior choice for managing commands within the shell.

Or else configuring Fish inside of DDEV globally or for a specific project.

**Comment by isalmanhaider** [comment]
Coming from #120 

Why not consider switching to Fish shell? It enhances the command-line experience by providing autocompletion for your most-used and recent commands. Additionally, Fish offers real-time feedback by highlighting command syntax to indicate whether commands are correct or recognized. This can significantly streamline workflows - while you can further amplify the experience with short aliases on top of Fish.

Fish shell is also particularly effective for Git operations. It can automatically suggest Git branches and commands as you type, which simplifies navigating through your repositories and enhances your version control workflow.

While ddev ssh is efficient, it typically requires typing out full commands repeatedly or cycling through command history with the arrow keys. By adopting Fish shell, you can reduce repetitive typing and increase productivity, making it a superior choice for managing commands within the shell.

Or else configuring Fish inside of DDEV globally or for a specific project.

**Comment by TravisCarden** [comment]
> Very low friction. I did not hit any blockers.

Great news, @alexb7217!

> Why not consider switching to Fish shell?

Thanks, @isalmanhaider. I think that's too aggressive right now. Bash is the still the safest bet for the largest number of people. Most folks probably don't even know what it is. And for the few that would really want it, I'm sure it wouldn't be a big effort to make the change--and document how.

**Comment by isalmanhaider** [comment]
@TravisCarden, I agree. In that case, we can pre-package Bash or Zsh with helpful plugins. This way, we provide a familiar environment for the majority of users while still enhancing their command-line experience. Thank you for your input, and I appreciate the consideration.

**Comment by alexb7217** [comment]
Task labeled "How to export and contribute Drupal config" -- am I misreading that or is contribute meant to be commit or import the config? @TravisCarden I should have more time to test and document now things have slowed down at work.

**Comment by boulaffasae** [comment]
** I encountered an issue because my version of DDEV was outdated, resulting in the error: "unknown command 'quick-start' for 'ddev'". After updating DDEV, everything worked perfectly.

Terminal stay active (because It open the website in a browser). I don't think we can avoid this

**Comment by TravisCarden** [comment]
### @alexb7217

> Okay, giving local setup a... "shot".

I failed to say anything before, but your pun deserves notice. 😝

> Task labeled "How to export and contribute Drupal config" -- am I misreading that or is contribute meant to be commit or import the config?

I've updated the wording to hopefully make it a little clearer... What I'm thinking is just that once people have done site building work locally, they may not know how to get the changes into a pull request. They'll need to know how to export config files to begin with, how to update existing recipes in the code, where to add new ones, etc. Thanks for your help with testing and documentation!



### @isalmanhaider

> In that case, we can pre-package Bash or Zsh with helpful plugins.

You might be interested in [creating your own DDEV add-on](https://ddev.readthedocs.io/en/stable/users/extend/additional-services/#creating-an-additional-service-for-ddev-get) for enhancements that don't become an official part of Starshot. I'm learning the process with an add-on for the related Experience Builder module (https://github.com/TravisCarden/ddev-drupal-xb-dev), and it's pretty neat.



### @boulaffasae 

> I encountered an issue because my version of DDEV was outdated, resulting in the error: "unknown command 'quick-start' for 'ddev'". After updating DDEV, everything worked perfectly.

Thanks. Most of those warnings are already known, per https://github.com/phenaproxima/starshot-prototype/issues/121. I haven't seen the last two before:

```
 [warning] The User-Agent must be set to use the Nominatim provider.
 [warning] Unable to geocode 'Kennedy Space Center Visitor Complex Space Commerce Way MERRITT ISLAND, FL 32953 US'.
```

Would you like to create an issue for them if they persists?

> Shouldn't we specify that the password is "admin" ?

Would you like to create an issue to discuss?

**Comment by phenaproxima** [comment]
Those warnings about Nominatim should be fixed in all circumstances as of #143, which I committed today.
